### PR TITLE
feat: add Notification CRD

### PR DESCRIPTION
**Pull request by xiao-jay**
Notification design docs https://jolly-antelope-3f4.notion.site/6736b535c9d443ae84ceddc2b7e99700

### CODE REVIEW

File: controllers/notification/main.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		os.Exit(1)
+	}
+	//+kubebuilder:scaffold:builder
+
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up ready check")
```
Comment (fanux) [inline controllers/notification/main.go:97]: ready check?

File: controllers/notification/main.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		os.Exit(1)
+	}
+
+	setupLog.Info("starting manager")
+	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+		setupLog.Error(err, "problem running manager")
```
Comment (fanux) [inline controllers/notification/main.go:103]: ???

Comment (fanux) [reply controllers/notification/main.go:103]: Failed to run manager.(maybe you need to use Google Translate)

File: controllers/notification/main.go (hunk 1 above)
Comment (xiao-jay) [reply controllers/notification/main.go:97]: I think this name is ok ,check pod is ready.

File: controllers/notification/main.go (hunk 2 above)
Comment (xiao-jay) [reply controllers/notification/main.go:103]: That is generate by kubebuilder.I'm going to submit a PR to kubebuilder

File: controllers/notification/README.md
Comment (zzjin) [inline controllers/notification/README.md:1]: Add some simple readme? just a few lines are ok.

File: controllers/notification/api/v1/notification_types.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+const Low Type = "Low"
+
+// NotificationSpec defines the desired state of Notification
+// UserName and whether read will be set in label,because set in label is ease to query
+type NotificationSpec struct {
+	Title      string `json:"title"`
+	Message    string `json:"message"`
+	TimeStamp  int64  `json:"timestamp"`
+	From       string `json:"from,omitempty"`
+	Importance Type   `json:"importance,omitempty"`
```
[the remark this answers is not in the document]
Comment (xiao-jay) [reply controllers/notification/api/v1/notification_types.go:36]: label,That will easy frontend to filler is_read and no_read notification CR

File: controllers/notification/main.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/labring/sealos/controllers/notification/controllers"
+
+	//+kubebuilder:scaffold:imports
+	notificationv1 "github.com/labring/sealos/controllers/notification/api/v1"
```
Comment (zzjin) [inline controllers/notification/main.go:38]: Seems notification need to move to common too, soon.

Comment (xiao-jay) [reply controllers/notification/main.go]: done

File: controllers/notification/api/v1/notification_types.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+const High Type = "High"
+const Medium Type = "Medium"
+const Low Type = "Low"
+
+// NotificationSpec defines the desired state of Notification
+// UserName and whether read will be set in label,because set in label is ease to query
+type NotificationSpec struct {
+	Title      string `json:"title"`
+	Message    string `json:"message"`
+	TimeStamp  int64  `json:"timestamp"`
```
[the remark this answers is not in the document]
Comment (xiao-jay) [reply controllers/notification/api/v1/notification_types.go]: done

File: controllers/notification/README.md
Comment (xiao-jay) [reply controllers/notification/README.md:1]: done

**Review by zzjin — APPROVED** [review]

**Pull request merged**